ProcessLib/Process.h

```diff
@@ -72,14 +72,59 @@ class Process
 	/// Process specific initialization called by initialize().
 	virtual void createLocalAssemblers() = 0;
 
-	/// Postprocessing after solve().
+	/// Preprocessing before starting assembly for new timestep.
+	virtual void preTimestep(GlobalVector const& x) {}
```

Comment:
They do change the internals. E.g., they could save some secondary variables from the previous timeste.